File: MWR/Source/MWRSolution/MWRDBModule/DBModule/TblMWEmployCtrl.cs
using System;
using System.Collections.Generic;
using System.Text;
using ComLib;
using ComLib.db;

namespace YRKJ.MWR
{
    public class TblMWEmployCtrl : BaseDataCtrl
    {
        public static bool QueryPage(DataCtrlInfo dcf, SqlWhere sw, int page, int pageSize, ref List<TblMWEmploy> itemList, ref string errMsg)
        {
            SqlQueryMng sqm = new SqlQueryMng();
            sqm.Condition.Where.AddWhere(sw);
            return QueryPage(dcf, sqm, page, pageSize, ref itemList, ref errMsg);
        }
 
        public static bool QueryPage(DataCtrlInfo dcf, SqlQueryMng sqm, int page, int pageSize,ref List<TblMWEmploy> itemList,ref string errMsg)
        {
            try
            {
                sqm.setQueryTableName(TblMWEmploy.getFormatTableName());
                string sql = sqm.getPageSql(page, pageSize);
                SqlCommonFn.DebugLog(sql);
                itemList = SqlDBMng.getInstance().query(sql, new TblMWEmploy(), sqm.getParamsArray());
                if (itemList.Count != 0)
                {
                    dcf.RowCount = itemList[0].TEM_COLUMN_COUNT;
                    dcf.PageCount = ComFn.getPageCount(pageSize, dcf.RowCount);
                }
            }
            catch (Exception e)
            {
                errMsg = e.Message;
                return false;
            }
            return true ;
        }

        public static bool QueryMore(DataCtrlInfo dcf, SqlWhere sw,ref List<TblMWEmploy> itemList,ref string errMsg)
        {
            SqlQueryMng sqm = new SqlQueryMng();
            sqm.Condition.Where.AddWhere(sw);
            return QueryMore(dcf, sqm,ref itemList,ref errMsg);
        }

        public static bool QueryMore(DataCtrlInfo dcf, SqlQueryMng sqm,ref List<TblMWEmploy> itemList,ref string errMsg)
        {
          
            try
            {
                sqm.setQueryTableName(TblMWEmploy.getFormatTableName());
                string sql = sqm.getSql();
                SqlCommonFn.DebugLog(sql);
                itemList = SqlDBMng.getInstance().query(sql, new TblMWEmploy(), sqm.getParamsArray());
            }
            catch (Exception e)
            {
                errMsg = e.Message;
                return false;
            }
            return true;
        }

        public static bool QueryOne(DataCtrlInfo dcf, SqlQueryMng sqm,ref TblMWEmploy item,ref string errMsg)
        {
            List<TblMWEmploy> itemList = null;
            if (!QueryMore(dcf, sqm, ref itemList, ref errMsg))
            {
                return false;
            }

            if (itemList.Count > 0)
            {
                item = itemList[0];
            }
            else
            {
                item = null;
            }

            return true;
        }

        public static bool QueryOne(DataCtrlInfo dcf, SqlWhere sw,ref TblMWEmploy item, ref string errMsg)
        {
            SqlQueryMng sqm = new SqlQueryMng();
            sqm.Condition.Where.AddWhere(sw);
            return QueryOne(dcf, sqm, ref item, ref errMsg);
        }

        public static bool Insert(DataCtrlInfo dcf, TblMWEmploy item, ref int count,ref string errMsg)
        {
            return Insert(dcf,
                item.EmpyCode,
                item.EmpyName,
                item.FuncGroupId,
                item.EmpyType,
                item.UserName,
                item.Password,
                item.Status,
                    ref count,
                    ref errMsg
                    );
        }

        public static bool Insert(DataCtrlInfo dcf,
            string empyCode,
            string empyName,
            int funcGroupId,
            string empyType,
            string userName,
            string password,
            string status,
                ref int _count,
                ref string _errMsg
                )
        {
            SqlUpdateMng sum = new SqlUpdateMng();
            sum.setQueryTableName(TblMWEmploy.getFormatTableName());
            sum.Add(TblMWEmploy.getEmpyCodeColumn(), empyCode);
            sum.Add(TblMWEmploy.getEmpyNameColumn(), empyName);
            sum.Add(TblMWEmploy.getFuncGroupIdColumn(), funcGroupId);
            sum.Add(TblMWEmploy.getEmpyTypeColumn(), empyType);
            sum.Add(TblMWEmploy.getUserNameColumn(), userName);
            sum.Add(TblMWEmploy.getPasswordColumn(), password);
            sum.Add(TblMWEmploy.getStatusColumn(), status);
            string sql = sum.getInsertSql();
            if (sql == null)
            {
                _errMsg = sum.ErrMsg;
                return false;
            }
            return doUpdateCtrl(dcf, sql,ref _count,ref _errMsg);
        }

        public static bool Update(DataCtrlInfo dcf, TblMWEmploy item, SqlWhere sw,ref int count,ref string errMsg)
        {
            SqlUpdateColumn suc = new SqlUpdateColumn();
            suc.Add(TblMWEmploy.getEmpyCodeColumn(), item.EmpyCode);
            suc.Add(TblMWEmploy.getEmpyNameColumn(), item.EmpyName);
            suc.Add(TblMWEmploy.getFuncGroupIdColumn(), item.FuncGroupId);
            suc.Add(TblMWEmploy.getEmpyTypeColumn(), item.EmpyType);
            suc.Add(TblMWEmploy.getUserNameColumn(), item.UserName);
            suc.Add(TblMWEmploy.getPasswordColumn(), item.Password);
            suc.Add(TblMWEmploy.getStatusColumn(), item.Status);
            return Update(dcf, suc, sw, ref count, ref errMsg);
        }

        public static bool Update(DataCtrlInfo dcf, TblMWEmploy item, SqlUpdateColumn suc, SqlWhere sw,ref int count,ref string errMsg)
        {
            if (suc.Columns != null)
                 SetUpdateColumnValue(suc, item);
            return Update(dcf, suc, sw, ref count, ref errMsg);
        }

        public static bool Update(DataCtrlInfo dcf, SqlUpdateColumn suc, SqlWhere sw,ref int count,ref string errMsg)
        {
            SqlUpdateMng sum = new SqlUpdateMng();
            sum.setQueryTableName(TblMWEmploy.getFormatTableName());
            string sql = sum.getUpdateSql(suc, sw);
            if (sql == null)
            {
                errMsg = sum.ErrMsg;
                return false;
            }
            return doUpdateCtrl(dcf, sql, ref count,ref errMsg);
        }

        public static bool Delete(DataCtrlInfo dcf, SqlWhere sw, ref int count, ref string errMsg)
        {
            SqlUpdateMng sum = new SqlUpdateMng();
            sum.setQueryTableName(TblMWEmploy.getFormatTableName());
            string sql = sum.getDeleteSql(sw);
            return doUpdateCtrl(dcf, sql, ref count,ref errMsg);
        }




    }
}

File: DemoApp/TblModel/TblMWEmployCtrl.java
package com.yrkj.ar.bean;
import java.util.List;

import com.yrkj.lib.ComFn;
import com.yrkj.lib.db.BaseDataCtrl;
import com.yrkj.lib.db.DataCtrlInfo;
import com.yrkj.lib.db.SqlCommonFn;
import com.yrkj.lib.db.SqlDBMng;
import com.yrkj.lib.db.SqlQueryMng;
import com.yrkj.lib.db.SqlUpdateColumn;
import com.yrkj.lib.db.SqlUpdateMng;
import com.yrkj.lib.db.SqlWhere;

public class TblMWEmployCtrl extends BaseDataCtrl {

    public static List<TblMWEmploy> QueryPage(DataCtrlInfo dcf,SqlQueryMng sqm,int page,int pageSize){
        List<TblMWEmploy> itemList = null;
        try{
            sqm.setQueryTableName(TblMWEmploy.getFormatTableName());
            String sql = sqm.getPageSql(page, pageSize);
            SqlCommonFn.DebugLog(sql);
            itemList = SqlDBMng.getInstance().query(sql, TblMWEmploy.class,sqm.getParamsArray());
            if(itemList.size() != 0){
                dcf.RowCount = itemList.get(0).TEM_COLUMN_COUNT;
                dcf.PageCount = ComFn.getPageCount(pageSize, dcf.RowCount);
            }
            dcf.set(true, "");
        }catch (Exception e) {
            dcf.set(false,e.getMessage());
        }
        return  itemList;
    }

    public static List<TblMWEmploy> QueryMore(SqlWhere sw) throws Exception {
        SqlQueryMng sqm = new SqlQueryMng();
        sqm.Condition.Where.AddWhere(sw);
        return QueryMore(sqm);
    }

    public static List<TblMWEmploy> QueryMore(DataCtrlInfo dcf,SqlWhere sw) {

        List<TblMWEmploy> itemList = null;
        try {
            itemList = QueryMore(sw);
            dcf.set(true, "");
        } catch (Exception e) {
            dcf.set(false, e.getMessage());
        }
        return itemList;
    }

    public static List<TblMWEmploy> QueryMore(SqlQueryMng sqm) throws Exception {
        sqm.setQueryTableName(TblMWEmploy.getFormatTableName());
        String sql = sqm.getSql();
        SqlCommonFn.DebugLog(sql);//System.out.println(sql);
        return SqlDBMng.getInstance().query(sql, TblMWEmploy.class,sqm.getParamsArray());
    }

    public static List<TblMWEmploy> QueryMore(DataCtrlInfo dcf,SqlQueryMng sqm){
        List<TblMWEmploy> itemList = null;
        try {
            itemList = QueryMore(sqm);
            dcf.set(true, "");
        } catch (Exception e) {
            dcf.set(false, e.getMessage());
        }
        return itemList;
    }

    public static TblMWEmploy QueryOne(SqlQueryMng sqm) throws Exception {
        List<TblMWEmploy> itemList = QueryMore(sqm);
        if(itemList.size() > 0){
            return itemList.get(0);
        }
        return null;
    }

    public static TblMWEmploy QueryOne(DataCtrlInfo dcf,SqlQueryMng sqm) {
        TblMWEmploy item = null;
        try {
            item = QueryOne(sqm);
            dcf.set(true, "");
        } catch (Exception e) {
            dcf.set(false, e.getMessage());
        }
        return item;
    }
    public static TblMWEmploy QueryOne(SqlWhere sw) throws Exception {
        SqlQueryMng sqm = new SqlQueryMng();
        sqm.Condition.Where.AddWhere(sw);
        return QueryOne(sqm);
    }

    public static TblMWEmploy QueryOne(DataCtrlInfo dcf,SqlWhere sw) {
        TblMWEmploy item = null;
        try {
            item = QueryOne(sw);
            dcf.set(true, "");
        } catch (Exception e) {
            dcf.set(false, e.getMessage());
        }
        return item;
    }

    public static int Insert(DataCtrlInfo dcf,TblMWEmploy item) {
        return Insert(dcf,
                item.EmpyCode,
                item.EmpyName,
                item.UserGroupId,
                item.EmpyType,
                item.UserName,
                item.Password
                );
    }


    public static int Insert(DataCtrlInfo dcf,
            String empyCode,
            String empyName,
            int userGroupId,
            String empyType,
            String userName,
            String password
            ) {
        SqlUpdateMng sum = new SqlUpdateMng();
        sum.setQueryTableName(TblMWEmploy.getFormatTableName());
        sum.Add(TblMWEmploy.getEmpyCodeColumn(), empyCode);
        sum.Add(TblMWEmploy.getEmpyNameColumn(), empyName);
        sum.Add(TblMWEmploy.getUserGroupIdColumn(), userGroupId);
        sum.Add(TblMWEmploy.getEmpyTypeColumn(), empyType);
        sum.Add(TblMWEmploy.getUserNameColumn(), userName);
        sum.Add(TblMWEmploy.getPasswordColumn(), password);
        String sql = sum.getInsertSql();
        return doUpdateCtrl(dcf,sql,sum.ErrMsg);
    }


    public static int Update(DataCtrlInfo dcf,TblMWEmploy item,SqlWhere sw) {
        SqlUpdateColumn suc = new SqlUpdateColumn();
        suc.Add(TblMWEmploy.getEmpyCodeColumn(), item.EmpyCode);
        suc.Add(TblMWEmploy.getEmpyNameColumn(), item.EmpyName);
        suc.Add(TblMWEmploy.getUserGroupIdColumn(), item.UserGroupId);
        suc.Add(TblMWEmploy.getEmpyTypeColumn(), item.EmpyType);
        suc.Add(TblMWEmploy.getUserNameColumn(), item.UserName);
        suc.Add(TblMWEmploy.getPasswordColumn(), item.Password);
        return Update(dcf,suc,sw);
    }

    public static int Update(DataCtrlInfo dcf,SqlUpdateColumn suc,SqlWhere sw) {
        SqlUpdateMng sum = new SqlUpdateMng();
        sum.setQueryTableName(TblMWEmploy.getFormatTableName());
        String sql = sum.getUpdateSql(suc,sw);
        return doUpdateCtrl(dcf,sql,sum.ErrMsg);
    }


    public static int Delete(DataCtrlInfo dcf,SqlWhere sw) {
        SqlUpdateMng sum = new SqlUpdateMng();
        sum.setQueryTableName(TblMWEmploy.getFormatTableName());
        String sql = sum.getDeleteSql(sw);
        return doUpdateCtrl(dcf,sql,sum.ErrMsg);
    }



}
